rust/src/treefile.rs

```diff
@@ -2755,7 +2755,12 @@ impl TreeComposeConfig {
     fn hasher_update(&self, hasher: &mut glib::Checksum) -> Result<()> {
         // don't use pretty mode to increase the chances of a stable serialization
         // https://github.com/projectatomic/rpm-ostree/pull/1865
-        hasher.update(serde_json::to_vec(self)?.as_slice());
+        let layers = serde_json::to_string(&self.base.ostree_layers.clone()).unwrap();
+        let mut j = serde_json::to_string(self)?;
+        let start = j.find(&layers).unwrap();
```

Comment:
Parsing JSON like this by searching for a substring seems very fragile to me.  I'd definitely prefer a more "structured" approach per above.  Can you try that out?

It's like editing JSON by doing e.g. `sed -i` on it instead of using `jq` for example.